sdk/search/Azure.Search.Documents/tests/Utilities/TestExtensions.cs

```diff
@@ -30,6 +30,22 @@ await foreach (T value in pageable)
             return default; // The compiler doesn't know that'll throw...
         }
 
+        /// <summary>
+        /// Get the first element in an async enumerable.
+        /// </summary>
+        /// <typeparam name="T">Type of elements.</typeparam>
+        /// <param name="sequence">The async enumerable.</param>
+        /// <returns>The first element.</returns>
+        public static async Task<T> FirstAsync<T>(this IAsyncEnumerable<T> sequence)
+        {
+            await foreach (T value in sequence)
+            {
+                return value;
+            }
+            Assert.Fail("No elements in sequence!");
```

Comment:
Why not just throw like `First` would then? It'll still fail the text, and works more generally (i.e. not just for tests)?

Reply:
Partly because I copied existing code from above.  Partly because it's in a class called TestExtensions which I don't ever anticipate for using for more than tests.  But mostly because I copied existing code.  :smile: